### PR TITLE
MINOR: Add transaction verification config to producerStateManager config

### PR DESCRIPTION
**Pull request by jolshan**
Originally part of https://github.com/apache/kafka/pull/13608/files. Since there are so many files changed, I decided to just pull this out into its own PR.

I have moved this config into producer state manager so it can be checked easily under the log lock when we are about to append.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/java/kafka/server/builders/LogManagerBuilder.java [hunk 1]
```diff
@@ -111,7 +111,12 @@ public LogManagerBuilder setMaxTransactionTimeoutMs(int maxTransactionTimeoutMs)
     }
 
     public LogManagerBuilder setMaxProducerIdExpirationMs(int maxProducerIdExpirationMs) {
-        this.producerStateManagerConfig = new ProducerStateManagerConfig(maxProducerIdExpirationMs);
+        this.producerStateManagerConfig = new ProducerStateManagerConfig(maxProducerIdExpirationMs, false);
```
Comment (dajac) [inline core/src/main/java/kafka/server/builders/LogManagerBuilder.java]: Should we remove `setMaxProducerIdExpirationMs` and use `setProducerStateManagerConfig` everywhere?

File: storage/src/main/java/org/apache/kafka/storage/internals/log/ProducerStateManagerConfig.java [hunk 2]
```diff
@@ -24,8 +24,11 @@ public class ProducerStateManagerConfig {
     public static final Set<String> RECONFIGURABLE_CONFIGS = Collections.singleton(PRODUCER_ID_EXPIRATION_MS);
     private volatile int producerIdExpirationMs;
 
-    public ProducerStateManagerConfig(int producerIdExpirationMs) {
+    private volatile boolean transactionVerificationEnabled;
```
Comment (dajac) [inline storage/src/main/java/org/apache/kafka/storage/internals/log/ProducerStateManagerConfig.java]: Do we plan to make it dynamic?

Comment (dajac) [inline storage/src/main/java/org/apache/kafka/storage/internals/log/ProducerStateManagerConfig.java]: small nit: I would put an empty line before `producerIdExpirationMs` and remove the one between `producerIdExpirationMs` and `transactionVerificationEnabled`.

**Review by dajac — COMMENTED** [review]
@jolshan Thanks for the PR. I left a few small comments.

File: storage/src/main/java/org/apache/kafka/storage/internals/log/ProducerStateManagerConfig.java (hunk 2 above)
Comment (jolshan) [reply storage/src/main/java/org/apache/kafka/storage/internals/log/ProducerStateManagerConfig.java]: I considered this, but the issue is that we currently start up a thread in startup to handle verification. I'm not sure we can do this dynamically.

Comment (jolshan) [reply storage/src/main/java/org/apache/kafka/storage/internals/log/ProducerStateManagerConfig.java]: can do

Comment (dajac) [reply storage/src/main/java/org/apache/kafka/storage/internals/log/ProducerStateManagerConfig.java]: In this case, the `volatile` is likely not required if we don't update it.

**Review by dajac — APPROVED** [review]
LGTM, thanks. We could perhaps update the title of the PR to be a little more precise.

**Comment by jolshan** [comment]
Checked the build -- streams upgrade tests had some issue that failed jdk 17 build.
There were also two test failures -- 

[Build / JDK 8 and Scala 2.12 / org.apache.kafka.tools.MetadataQuorumCommandErrorTest.testRelativeTimeMs()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13770/3/testReport/junit/org.apache.kafka.tools/MetadataQuorumCommandErrorTest/Build___JDK_8_and_Scala_2_12___testRelativeTimeMs__/)
[Build / JDK 11 and Scala 2.13 / org.apache.kafka.streams.integration.SmokeTestDriverIntegrationTest.[1] true](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13770/3/testReport/junit/org.apache.kafka.streams.integration/SmokeTestDriverIntegrationTest/Build___JDK_11_and_Scala_2_13____1__true/)

Given the nature of this change -- basically a no-op except for some test files that all passed. I will go ahead and merge.

**Pull request merged**